Backend/router/v1/report.ctrl.js

```diff
@@ -1,120 +1,122 @@
+    what: req.body.data[0].what,
+    location: req.body.data[0].location,
+    time: req.body.data[0].time,
+    who: req.body.data[0].who,
+    details: req.body.data[0].details,
+    userID: req.user[0].dataValues.id
+  }).then((result) => {
+    res.status(201);
+    res.json({id: result.dataValues.id});
+  }).catch((err) => {
+    res.json({msg: "Error: failed to create report"});
+  });
 };
 
 const showReportList = (req, res) => {
-  if (req.isAuthenticated()) {
-    models.report.findAll({
-      where: {
-        userID: req.user[0].dataValues.id
-      }
-    }).then((result) => {
-      res.status(200);
-      res.json(result);
-    }).catch((err) => {});
-  } else {
-    res.status(401)
-    res.json({
-      msg: "get report failed"
-    })
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
+    return ;
   }
+  models.report.findAll({
+    where: {
+      userID: req.user[0].dataValues.id
+    }
+  }).then((result) => {
+    res.status(200);
+    res.json(result);
+  }).catch((err) => {
+    res.json({msg: "Error: failed to load all reports"});
+  });
 };
 
 const findReport = (req, res) => {
-  if (req.isAuthenticated()) {
-    const reportId = parseInt(req.query.reportId);
-
-    if (Number.isNaN(reportId)) {
-      return res.status(400).end()
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
+    return ;
+  }
+  const reportId = parseInt(req.query.reportId);
+  if (Number.isNaN(reportId)) {
+    return res.status(400).end();
+  }
+  models.report.findAll({
+    where: {
+      id: reportId
     }
-    models.report.findAll({
-      where: {
-        id: reportId
-      }
-    }).then((result) => {
+  }).then ((result) => {
+    if (result == 0) {
+      res.json({msg: "The reportId does not exist"});
+    } else {
       res.json(result);
-    }).catch((err) => {});
-  } else {
-    res.json({
-      msg: "Can't find this report"
-    });
-  }
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to load this report"});
+  });
 };
 
 const deleteReport = (req, res) => {
-  const reportId = parseInt(req.params.reportId);
-  //When ID is not a number
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
+    return ;
+  }
+  const reportId = parseInt(req.query.reportId);
   if (Number.isNaN(reportId)) {
-    return res.status(400).end()
+    return res.status(400).end();
   }
   models.report.destroy({
     where: {
       id: reportId
     }
   }).then((result) => {
-    res.status(204).json(result);  //no content
-  }).catch((err) => {});
+    if (result == 0) {
+      res.json({msg: "The reportId does not exist"});
+    } else {
+      res.status(204).end();
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to delete this report"});
+  });
 };
 
 const editReport = (req, res) => {
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
```

Comment:
Can you update the error message so it's more consistent with "Error: failed to delete this report"?

Reply:
I don't know what error is caught and I think this is enough. @shberry